coding/transform.go

```diff
@@ -5,26 +5,31 @@ import (
 
 	mc "github.com/jbenet/go-multicodec"
 	mcjson "github.com/jbenet/go-multicodec/json"
+	mccbor "github.com/jbenet/go-multicodec/cbor"
 	ipld "github.com/ipfs/go-ipld"
 )
 
-type codec struct {
+type transformCodec struct {
 	mc.Multicodec
 }
 
-type decoder struct {
+type transformDecoder struct {
 	mc.Decoder
 }
 
-func jsonMulticodec() mc.Multicodec {
-	return &codec{mcjson.Multicodec(false)}
+func JsonMulticodec() mc.Multicodec {
+	return &transformCodec{mcjson.Multicodec(false)}
 }
 
-func (c *codec) Decoder(r io.Reader) mc.Decoder {
-	return &decoder{ c.Multicodec.Decoder(r) }
+func CborMulticodec() mc.Multicodec {
+	return &transformCodec{mccbor.Multicodec()}
```

Comment:
ah this means cbor stuff is now walked twice, right? once by the cbor decoder, once again by the transform here? (this sounds unavoidable, as your test points out, unless we go back to ugorji codec which allowed us to set the type to construct objects with, or do something similar to that)